app/javascript/controllers/favorite_controller.js

```diff
@@ -5,7 +5,29 @@ export default class extends HeartController {
     id: Number,
   }
 
+  connect() {
+    // access controller from child element
+    this.element[this.identifier] = this
```

Comment:
It " hangs a reference to the Stimulus controller instance off the DOM element that has the same name as the controller itself".

Since there are multiple instances of same asset/mp3, this line allows you to access the other controllers to toggle and keep the favorite count in sync.

Technique was taken from this article:
https://dev.to/leastbad/the-best-one-line-stimulus-power-move-2o90

Reply:
Thanks for the article! I've used `getControllerForElementAndIdentifier` elsewhere, so was curious how this differed, whether events would be more appropriate, etc.